rust/gsc/src/main.rs

```diff
@@ -44,13 +337,21 @@ async fn main() -> Result<(), Box<dyn std::error::Error>> {
     let state = Arc::new(connection);
 
     // build our application with a single route
-    let app = Router::new().route(
-        "/job",
-        post({
-            let shared_state = state.clone();
-            move |body| add_job(body, shared_state)
-        }),
-    );
+    let app = Router::new()
+        .route(
+            "/job",
+            post({
+                let shared_state = state.clone();
+                move |body| add_job(body, shared_state)
+            }),
+        )
+        .route(
+            "/job/matching",
```

Comment:
Feel free to rename this as you see fit we don't have to go with this route name

Reply:
I'm very ok with whatever route name you want. My gripe is with other RESTful stuff.